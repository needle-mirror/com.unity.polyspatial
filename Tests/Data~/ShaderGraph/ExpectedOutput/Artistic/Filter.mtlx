<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <dot name = "_1" type = "float">
        <input name = "in" type = "float" nodename = "_2" />
    </dot>
    <clamp name = "_2" type = "float">
        <input name = "in" type = "float" nodename = "_3" />
    </clamp>
    <add name = "_3" type = "float">
        <input name = "in1" type = "float" nodename = "_4" />
        <input name = "in2" type = "float" nodename = "_8" />
    </add>
    <multiply name = "_4" type = "float">
        <input name = "in1" type = "float" nodename = "_5" />
        <input name = "in2" type = "float" nodename = "_6" />
    </multiply>
    <dot name = "_5" type = "float">
        <input name = "in" type = "float" nodename = "_11" />
    </dot>
    <add name = "_6" type = "float">
        <input name = "in1" type = "float" nodename = "_7" />
        <input name = "in2" type = "float" value = "1" />
    </add>
    <dot name = "_7" type = "float">
        <input name = "in" type = "float" value = "1" />
    </dot>
    <multiply name = "_8" type = "float">
        <input name = "in1" type = "float" nodename = "_9" />
        <input name = "in2" type = "float" nodename = "_7" />
    </multiply>
    <subtract name = "_9" type = "float">
        <input name = "in1" type = "float" nodename = "_10" />
        <input name = "in2" type = "float" value = "1" />
    </subtract>
    <dot name = "_10" type = "float">
        <input name = "in" type = "float" value = "0.5" />
    </dot>
    <constant name = "_SinTime" type = "vector4">
        <input name = "value" type = "vector4" value = "0,0,0,0" />
    </constant>
    <swizzle name = "_11" type = "float">
        <input name = "channels" type = "string" value = "w" />
        <input name = "in" type = "vector4" nodename = "_SinTime" />
    </swizzle>
    <GeometryModification name = "_12" type = "vertexshader">
    </GeometryModification>
    <realitykit_unlit name = "_13" type = "surfaceshader">
        <input name = "applyPostProcessToneMap" type = "boolean" value = "false" />
        <input name = "color" type = "color3" nodename = "_14" />
    </realitykit_unlit>
    <surfacematerial name = "USD_Filter" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_12" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_13" />
    </surfacematerial>
    <swizzle name = "_14" type = "color3">
        <input name = "in" type = "float" nodename = "_1" />
        <input name = "channels" type = "string" value = "xxx" />
    </swizzle>
</materialx>
