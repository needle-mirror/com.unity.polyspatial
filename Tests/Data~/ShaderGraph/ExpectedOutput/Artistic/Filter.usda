#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Filter"
        {
            float4 inputs:_SinTime = (0, 0, 0, 0)

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_14.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_swizzle_float_color3"
                float inputs:in.connect = <../_1.outputs:out>
                string inputs:channels = "xxx"
                color3f outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_2.outputs:out>
                float outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_clamp_float"
                float inputs:in.connect = <../_3.outputs:out>
                float outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_4.outputs:out>
                float inputs:in2.connect = <../_8.outputs:out>
                float outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_5.outputs:out>
                float inputs:in2.connect = <../_6.outputs:out>
                float outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_11.outputs:out>
                float outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                string inputs:channels = "w"
                float4 inputs:in.connect = <../.inputs:_SinTime>
                float outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_9.outputs:out>
                float inputs:in2.connect = <../_7.outputs:out>
                float outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_10.outputs:out>
                float inputs:in2 = 1
                float outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 0.5
                float outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_7.outputs:out>
                float inputs:in2 = 1
                float outputs:out
            }
        }
    }
}
