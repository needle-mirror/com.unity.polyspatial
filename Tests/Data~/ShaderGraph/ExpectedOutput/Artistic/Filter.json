{"properties":[{"name":"_SinTime","type":3}]}